### PR TITLE
Chisel release format V1

### PR DESCRIPTION
**Pull request by letFunny**
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?

-----
Introduce support for chisel release format "v1", maintaining compatibility with "chisel-v1". The difference is in `chisel-v1` the attribute for public keys is named `v1-public-keys`, while in `v1` it is called `public-keys`. The format `v1` will serve to break compatibility with versions that predate GPG verification for downloading releases.

### CODE REVIEW

File: internal/setup/releaseChiselV1.go [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+	"bytes"
+	"fmt"
+
+	"golang.org/x/crypto/openpgp/packet"
+	"gopkg.in/yaml.v3"
+
+	"github.com/canonical/chisel/internal/pgputil"
+)
+
+func ParseReleaseChiselV1(baseDir, filePath string, data []byte) (*Release, error) {
```
Comment (rebornplusplus) [inline internal/setup/releaseChiselV1.go:13]: nit: Does this need to be exported?

File: internal/setup/releaseV1.go [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+	"bytes"
+	"fmt"
+
+	"golang.org/x/crypto/openpgp/packet"
+	"gopkg.in/yaml.v3"
+
+	"github.com/canonical/chisel/internal/pgputil"
+)
+
+func ParseReleaseV1(baseDir, filePath string, data []byte) (*Release, error) {
```
Comment (rebornplusplus) [inline internal/setup/releaseV1.go:13]: nit: same comment as above. Does this need to be exported?

**Review by rebornplusplus — COMMENTED** [review]
Looks good to me overall. Left a few nitpicks.

File: internal/setup/releaseChiselV1.go (hunk 1 above)
Comment (letFunny) [reply internal/setup/releaseChiselV1.go:13]: You are totally right!

File: internal/setup/releaseV1.go (hunk 2 above)
Comment (letFunny) [reply internal/setup/releaseV1.go]: You are right!

**Pull request closed without merging**